Add draft governance ZIPs (#989)
zcash/zips: zips/draft-ecc-community-and-coinholder.md

    ZIP: XXX
    Title: Community and Coinholder Funding Model
    Owners: Josh Swihart <josh@electriccoin.co>
    Credits: Daira-Emma Hopwood
             Kris Nuttycombe
             Jack Grigg

>              Tatyana Peacemonger

Reviewer: Add Alex Bornstein and Jason McGee to credits

Reviewer 2: https://github.com/zcash/zips/pull/994

# Non-requirements

* Any changes to the ZCG or its governance, such as its expansion to include more members, may be desired but are specifically outside this proposal’s scope.
* Any changes to Zcash protocol governance, specifically what changes are made to consensus node software, are outside this proposal’s scope.

# Specification

This proposal empowers both the community (through ZCG) and coin holders to independently determine what, if anything, should be funded through development fund grants.

The funding streams described below will be defined in a new revision of ZIP 214 [^zip-0214].

## Zcash Community Grants

A funding stream will be established for Zcash Community Grants, consisting of 8% of the block subsidy, and subject to all of the same rules currently specified in ZIP 1015 [^zip-1015-zcg].

This funding stream will start on expiry of the existing ``FS_FPF_ZCG`` funding stream [^zip-1015-funding-streams], and last for a further 1,260,000 blocks (approximately 3 years), ending at Zcash's 3rd halving.

## Coinholder-Controlled Fund

> A pool of multisig-controlled funds, seeded from the existing contents of the ZIP 1015 Deferred Dev Fund Lockbox [^zip-1015-lockbox] and supplemented with a funding stream consising of 12% of the block subsidy for the same time period as the Zcash Community Grants stream, forms a new Coinholder-Controlled Fund. The mechanisms for the creation and management of this fund are described by the Deferred Dev Fund Lockbox Disbursement proposal [^draft-ecc-lockbox-disbursement]. This proposal sets the $\mathsf{stream\_value}$ parameter of that proposal to 12%, and the $\mathsf{stream\_end\_height}$ parameter to mainnet block height 4406400, equal to that of the Zcash Community Grants stream so that both streams end at Zcash's 3rd halving.

Author: s/mainnet/Mainnet/

Reviewer 2: Addressed in https://github.com/zcash/zips/pull/994

 
### Requirements on use of the Coinholder-Controlled Fund

> The Key-Holder Organizations SHALL be bound by a legal agreement to only use funds held in the Coinholder-Controlled Fund according to the specifications in this ZIP, expressed in suitable legal language. In particular, all requirements on the use of Deferred Dev Fund Lockbox funds in ZIP 1015 [^zip-1015-lockbox] MUST be followed for Coinholder-Controlled Fund.

Author: s/for /for the /

Reviewer 2: Addressed in https://github.com/zcash/zips/pull/994

The Key-Holder Organizations collectively administer the Coinholder-Controlled Fund based on decisions taken by coinholder voting, following a model decided by the community and specified in another ZIP [TBD].

### Disbursement process

1. Anyone can submit a grant application via a process agreed upon by the Key-Holder Organizations.

> 2. Grant applications with total below USD 500,000 (or equivalent in another currency) are directed to ZCG.

Reviewer: Remove " Grant applications with total below USD 500,000 (or equivalent in another currency) are directed to ZCG."

> 3. For grant applications above this threshold, a 30-day community review and feedback period will start.

Reviewer 2: https://github.com/zcash/zips/pull/994

4. If a grant application is not [vetoed](#veto-process) and proceeds to a coinholder vote according to the agreed process, then coinholders will be asked to vote on it.
5. If the vote passes, then as payments are scheduled for the grant, then (subject to the [Veto process]) the Key-Holder Organizations SHOULD sign the corresponding transactions for disbursement from the Coinholder-Controlled Fund.

For coinholder votes, a minimum of 420,000 ZEC (2% of the total eventual supply) MUST be voted, with a simple majority cast in favor, for a grant proposal to be approved. Upon approval, the grants are paid to the recipient per the terms of the grant proposal. In the case that multiple grant proposals are submitted that are in competition with one another, a single winner will be selected by holding a separate coinholder vote for each proposal, with the approved proposal having the highest total ZEC value committed in support being the winner. It is the responsibility of the Key-Holder Organizations to determine whether proposals are in competition with one another when organizing coinholder votes.

> Each coinholder vote (including in cases where multiple grants are in competition) is independent, and coins used in one vote are also allowed to be used in another; this approach is necessary to avoid vote-splitting scenarios that can result in an option being selected that achieves only a plurality (not a majority) of coinholder support.

Author: Is it clear enough to non-voting-system-nerds what this means, or do we need more explanation?

Reviewer 2: As someone who's not in the set of non-voting-system-nerds, I can't judge!

If a grant is vetoed after passing a coinholder vote, then payments for it MUST be stopped. This is expected to be an unusual situation that would only occur if new adverse information came to light, or in the case of a change in the law or unanticipated legal proceedings.

The Key-holder Organizations cannot veto coinholder rejection of a proposal.

Vetos are intended for exceptional cases and SHOULD be accompanied by a thorough rationale.

## Administrative obligations and constraints

All provisions of ZIP 1015 imposing obligations and constraints on Bootstrap Project, Electric Coin Company, Zcash Foundation, Financial Privacy Foundation, the ZCG Committee, and grant recipients relating to the previous ``FS_FPF_ZCG`` funding stream, SHALL continue in effect for Zcash Community Grants.

These obligations and constraints will be extended to all Key-Holder Organizations in respect of the Coinholder-Controlled Fund.

The provisions after the first paragraph of the section "Zcash Community Grants (ZCG)" also apply to the Key-Holder Organizations' administration of the Coinholder-Controlled Fund, with coinholder voting replacing the role of the ZCG Committee.

> Note: Nothing forces developers of Zcash consensus node software to implement any particular proposal. The aim of a process specification like this one is only to social consensus. It fundamentally cannot affect the autonomy of developers of Zcash consensus node software to publish (or not) the software they want to publish, or the autonomy of node operators to run (or not) the software they want to run.

Author: "only to indicate social consensus"? or "establish"?

## Security precautions

The Key-Holder Organizations and the ZCG Committee MUST take appropriate precautions to safeguard funds from theft or accidental loss. Any theft or loss of funds, or any loss or compromise of key material MUST be reported to the Zcash community as promptly as possible after applying necessary mitigations.

## Testnet-specific considerations

In order to allow the mechanism and process for coinholder voting to be tested, this process SHOULD be rehearsed on Testnet. The threshold of 420,000 ZEC applied to Mainnet coinholder votes does not apply to these rehearsals.

# Open questions

> * Is a 3-of-5 threshold appropriate? What should the other two Key-Holder Organizations be?

Reviewer 2: https://github.com/zcash/zips/pull/994

* Is 420,000 ZEC the right threshold for coinholder votes?

> * Is USD 500,000 the right threshold for grants to be controlled by coinholder vote rather than ZCG?

Reviewer: Remove

Reviewer 2: https://github.com/zcash/zips/pull/994

# Motivation

If no action is taken, in November 2025 funds from block subsidies will stop being directed to Zcash Community Grants [^zip-1015-zcg] and to the Deferred Dev Fund Lockbox established by ZIP 2001 [^zip-2001]. If the block subsidies stop, it will risk a gap in funding of Zcash development organisations, either via ZCG grants or via potential future disbursements from the Deferred Dev Fund Lockbox.

This proposal aims to:
* decentralize decision-making,
* hold stakeholders accountable for outcomes,
* be dynamic enough to allow for change, and
* provide clarity on decision-making.

It would immediately increase coin-holders’ voice, minimize governance confusion, and simplify decision-making.

# Requirements

* There is a well-defined, publically agreed, process for evaluation and community feedback on grant proposals.
* Funds are held in a multisig resistant to compromise of some of the parties' keys, up to a given threshold.
* No single party's non-cooperation or loss of keys is able to cause any Protocol-defined Ecosystem Funding to be locked up unrecoverably.

> * During the period of this proposal, the block rewards must be distributed as described in the [Abstract] above.

Reviewer 2: The word "distributed" in this sentence seems overloaded to me. Does this refer to the 8%/12% split, or the sentence that follows (which doesn't provide enough clarity to qualify as a requirement)?

> * The funds from the 8% and 12% funding streams will be usable immediately.

Reviewer 2: This requirement precludes the use of a buffering approach. Is that intended?

Author: This wasn't true of the 12% stream (and in Option 1 there isn't any such stream). So "and 12%" has been deleted and this comment no longer applies. The 8% will continue to be provided via a funding stream without buffering.

> * The funds in the Deferred Dev Fund Lockbox will be usable immediately.

Reviewer 2: "immediately" is not well-defined. At best, this should be "as of the activation height of the protocol change specified by this ZIP".

Author: Changed to "usable immediately on activation of this ZIP".

> * Any use of Deferred Dev Fund Lockbox funds is consistent with the purpose specified in [^zip-1015-lockbox] of "funding grants to ecosystem participants".

Reviewer 2: ```suggestion
* Any use of Deferred Dev Fund Lockbox funds must be consistent with the purpose specified in [^zip-1015-lockbox] of "funding grants to ecosystem participants".
```

Author: The other requirements use "is", "will be", etc. They should all be changed or none of them changed. I think it reads better as-is.

# Non-requirements

Any changes to the ZCG or its governance, such as its expansion to include more members, may be desired but are specifically outside this proposal’s scope.

# Specification

This proposal empowers both the community (through ZCG) and coin holders to independently determine what, if anything, should be funded through development fund grants.

The funding streams described below will be defined in a new revision of ZIP 214 [^zip-0214].

## Zcash Community Grants

A funding stream will be established for Zcash Community Grants, consisting of 8% of the block subsidy, and subject to all of the same rules currently specified in ZIP 1015 [^zip-1015-zcg].

This funding stream will start on expiry of the existing ``FS_FPF_ZCG`` funding stream [^zip-1015-funding-streams], and last for a further 1,260,000 blocks (approximately 3 years), ending at Zcash's 3rd halving.

## Coin-holder-controlled Fund

### One-time lockbox disbursal

> The coinbase transaction of the activation block of this ZIP MUST include an additional output to a 3-of-5 P2SH multisig with keys held by the Zcash Foundation, the Electric Coin Company, Shielded Labs, and two other organizations yet to be decided ("Key-holder Organizations").

Reviewer 2: nit: I think that `Key-Holder`, `Coin-Holder`, and `Coin-Holder Controlled` should have all words capitalized; to capitalize just the first word of the hyphenated phrase looks wrong to me.
```suggestion
The coinbase transaction of the activation block of this ZIP MUST include an additional output to a 3-of-5 P2SH multisig with keys held by the Zcash Foundation, the Electric Coin Company, Shielded Labs, and two other organizations yet to be decided ("Key-Holder Organizations").
```

If accepted, apply this suggestion throughout.

This stream will replace the existing ``FS_DEFERRED`` lockbox funding stream [^zip-1015-funding-streams]. That is, the end height of ``FS_DEFERRED`` will be changed to the activation height of this ZIP. The new ``FS_COINHOLDER`` stream will start at the same height and last until the 3rd halving.

### Option 3: Periodic lockbox disbursement

This would require a mechanism to pay from the Deferred Dev Fund Lockbox to a 3-of-5 P2SH multisig with keys held by the same Key-holder Organizations as above, at periodic intervals. [TBD schedule and period.]

### Requirements on use of the Coin-holder-controlled Fund

The Key-Holder Organizations would be bound by a legal agreement to only use funds held in the Coin-holder-controlled Fund according to the specifications in this ZIP, expressed in suitable legal language. In particular, all requirements on the use of Deferred Dev Fund Lockbox funds in ZIP 1015 [^zip-1015-lockbox] MUST be followed for the funds obtained from the one-time disbursal.

> The Financial Privacy Foundation (FPF) would administer the Coin-holder-controlled Fund based on decisions taken by coin-holder voting, following a model decided by the community and specified in another ZIP [TBD].

Reviewer 2: Note from Josh: 

>  FPF will not be involved. Some org (TBD) will need to be designated to conduct polling.

Author: This now reads "The Key-Holder Organizations would collectively administer the Coinholder-Controlled Fund based on decisions taken by coinholder voting, ..."

### Disbursement process

> 1. Anyone can submit a grant application via a process agreed between the FPF, the Zcash Foundation, and the Electric Coin Company.

Reviewer 2: Adjust to remove the FPF; probably this should be:
```suggestion
1. Anyone can submit a grant application via a process agreed upon by the Key-Holder Organizations.
```

2. Grant applications with total below USD 500,000 (or equivalent in another currency) would be directed to ZCG.

> 3. For grant applications above this threshold, a 30-day community review and feedback period would start.

Reviewer 2: ```suggestion
3. For grant applications above this threshold, a 30-day community review and feedback period will start.
```

> 4. If the grant applications should proceed to a coin-holder vote according to the agreed process, then coin-holders would be asked to vote on it.

Reviewer 2: ```suggestion
4. If a grant application proceeds to a coin-holder vote according to the agreed process, then coin-holders will be asked to vote on it.
```
This doesn't specify a timeline; does the vote take place at the end of the 30 day window? As written, this looks like it allows a vote to be deferred indefinitely.

Reviewer 2: Also: at present this doesn't define a coin-holder voting mechanism. Using Hanh's coin voting mechanism restricts votes to funds held in the Orchard pool; if that's intended, then it should be explicitly specified; however, that might be construed to contradict https://github.com/zcash/zips/pull/989/files#diff-8528553152727684ca93a225882ceed8e57a309ae833f5f517f0acbf573cd61aR129 which says that organizations cannot be restricted from voting their coins; if some barrier (such as lack of server-grade HSM support for Orchard) prevents an organization from safely voting their coins, is that a problem?

Author: > This doesn't specify a timeline; does the vote take place at the end of the 30 day window?

As currently written, the "process agreed upon by the Key-Holder Organizations" decides that. So it's not ambiguous, but we might want to tighten it up.

> 5. If the vote passes, then as payments are scheduled for the grant, then the FPF, the Zcash Foundation, and the Electric Coin Company SHOULD sign the corresponding transactions for disbursement from the Coin-holder-controlled Fund.

Reviewer 2: ```suggestion
5. In order to ensure that subversion of the Coin-holder voting mechanism cannot result in draining of the Coin-holder-controlled Fund, and to ensure there is no disbursement of funds in a manner that is antithetical to the values of the Zcash community, the Key-holder Organizations MAY veto coin-holder approval via public announcement within 10 days of the tabulation of the coin-holder poll. If such a veto is exercised, the funds specified in the vetoed grant will remain at rest. The Key-holder Organizations MAY NOT veto coinholder rejection of a proposal.
6. If the coinholder vote approves the grant and no veto is exercised, then the Key-holder Organizations SHOULD sign transactions for disbursement from the Coin-holder-controlled Fund according to the schedule proposed by the grant. 
```

Author: @str4d, @nuttycom and I discussed cases where there is disagreement between the Key-Holder Organizations so that they cannot come to a collective decision on a veto. In that case, I suggested that it should be sufficient for at least two of the Key-Holder Organizations to agree on a veto. (A threshold of one presents too much risk of a single org becoming an obstruction; a threshold of three is too high since this is a veto mechanism, not a voting mechanism.)

Author: The current wording reflects that there are two criteria for a veto: legal or reporting requirements which have a threshold of 1, and other objections which have a threshold of 2.

For coin holder votes, a minimum of 420,000 ZEC (2% of the total eventual supply) MUST be voted, with a simple majority cast in favor, for a grant proposal to be approved. Upon approval, the grants would be paid to the recipient per the terms of the grant proposal.

> Coin holders would not be obligated to fund grants and MAY leave the funds at rest.

Reviewer 2: This sentence seems redundant to me, but if it's kept in I think it should be clarified:
```suggestion
Coin holders are not obligated to fund any grants and MAY leave the funds at rest.
```

> No organizations or individuals would be restricted from voting their coins.

Reviewer 2: This seems like it belongs in the `Requirements` section. It's not really specification.
```suggestion
No organizations or individuals may be restricted from voting their coins.
```

## Administrative obligations and constraints

All provisions of ZIP 1015 imposing obligations and constraints on Bootstrap Project, Electric Coin Company, Zcash Foundation, Financial Privacy Foundation, the ZCG Committee, and grant recipients relating to the previous ``FS_FPF_ZCG`` funding stream, SHALL continue in effect for *both* Zcash Community Grants and the Coin-holder-controlled Fund.

These obligations and constraints will be extended to Shielded Labs and the other Key-holder Organizations in respect of the Coin-holder-controlled Fund.

The provisions after the first paragraph of the section "Zcash Community Grants (ZCG)" also apply to FPF's administration of the Coin-holder-controlled Fund, with coin-holder voting replacing the role of the ZCG Committee.

For the avoidance of doubt, this includes the following:

* Decisions on use of the Coin-holder-controlled Fund taken by coin-holder vote require no approval from the FPF Board, but are subject to veto if FPF judges them to violate Cayman law or the FPF's reporting requirements and other (current or future) obligations under the Cayman Islands' Companies Act (2023 Revision) and Foundation Companies Law, 2017.

> * FPF SHALL be contractually required to recognize the Coin-holder-controlled Fund as a Restricted Fund donation under the above constraints, and keep separate accounting of its balance and usage under its Transparency and Accountability obligations defined in ZIP 1015 [^zip-1015-transparency-and-accountability].

Reviewer 2: If the FPF is not involved (as mentioned by Josh above) do these clauses need to be retained, or should they be modified to refer to the Key-Holder Organizations?

Author: They don't need to be retained given the section on Veto process.

> Coin-holders SHOULD take account of the same factors considered by the ZCG Committee (described in points 2, 3, and 4 of [^zip-1015-zcg]) in deciding whether to fund a grant. If any contentious issue arises in connection with a grant (such as milestones not being met), or periodically for grants of indefinite duration, FPF SHOULD hold additional coin-holder votes to determine whether funding should continue.

Reviewer 2: Does this also need to be modified to refer to the Key-Holder Organizations instead of the FPF?
```suggestion
Coin-holders SHOULD take account of the same factors considered by the ZCG Committee (described in points 2, 3, and 4 of [^zip-1015-zcg]) in deciding whether to fund a grant. If any contentious issue arises in connection with a grant (such as milestones not being met), or periodically for grants of indefinite duration, the Key-Holder Organizations SHOULD administer additional coin-holder votes to determine whether funding should continue.
```

* The funds in the Deferred Dev Fund Lockbox will be usable immediately on activation of this ZIP.
* Any use of Deferred Dev Fund Lockbox funds is consistent with the purpose specified in [^zip-1015-lockbox] of "funding grants to ecosystem participants".

> 

Author: It's not important (and makes no difference to rendering), but we generally leave two lines between top-level sections.

> 

Author: TODO: check whether this is correct for Testnet.

## Coinholder-Controlled Fund

> The pool of multisig-controlled funds described as the Zcash Development Fund by [^draft-ecc-lockbox-distribution] forms a new Coinholder-Controlled Fund.

Author: Calling it the Zcash Development Fund is confusing because in ZIPs 1014 and 1015, that included the ZCG Slice.

Reviewer 2: I don't know how to generalize this name, because in the community-and-coinholder proposal this fund receives 12% of the block reward, and in the ZBloc proposal it receives 20%.

Reviewer 2: Figured it out.

4. If a grant application is not [vetoed](#veto-process) and proceeds to a coinholder vote according to the agreed process, then coinholders will be asked to vote on it.
5. If the vote passes, then as payments are scheduled for the grant, then (subject to the [Veto process]) the Key-Holder Organizations SHOULD sign the corresponding transactions for disbursement from the Coinholder-Controlled Fund.

> For coinholder votes, a minimum of 420,000 ZEC (2% of the total eventual supply) MUST be voted, with a simple majority cast in favor, for a grant proposal to be approved. Upon approval, the grants are paid to the recipient per the terms of the grant proposal. In the case that multiple grant proposals are submitted that are in competition with one another, a single winner will be selected by holding a separate coinholder vote for each proposal, with the approved proposal having the highest total ZEC value committed in support being the winner.

Reviewer 2: Good point; the Key-Holder Organizations (since they're running the coinholder votes) will need to make this determination.
